crates/ruff_python_formatter/tests/snapshots/format@statement__match.py.snap

```diff
@@ -267,23 +285,40 @@ match pattern_comments:
 match pattern_comments:
     case (
         # leading
-        x as NOT_YET_IMPLEMENTED_PatternMatchAs
+        only_leading
     ):
         pass
 
 
 match pattern_comments:
     case (
         # leading
-        x as NOT_YET_IMPLEMENTED_PatternMatchAs  # trailing 1
+        leading_and_trailing  # trailing 1
         # trailing 2
         # trailing 3
     ):
         pass
 
 
 match pattern_comments:
-    case (x as NOT_YET_IMPLEMENTED_PatternMatchAs):
+    case (no_comments):
+        pass
+
+
+match pattern_comments:
+    case (
+        # 1
+        pattern as name  # 2
+        # 3
+        # 4
+        # 5  # 6
+        # 7
+    ):
+        pass
```

Comment:
This might need to be handled in a different way as there seems to be 2 entries created for `PatternMatchAs` node comments where the first entry contains 1, 6, 7 and the second contains 2, 3, 4, 5.

```rust
{
    Node {
        kind: PatternMatchAs,
        range: 55..120,
        source: `pattern  # 2⏎`,
    }: {
        "leading": [
            SourceComment {
                text: "# 1",
                position: OwnLine,
                formatted: true,
            },
        ],
        "dangling": [],
        "trailing": [
            SourceComment {
                text: "# 6",
                position: EndOfLine,
                formatted: true,
            },
            SourceComment {
                text: "# 7",
                position: OwnLine,
                formatted: true,
            },
        ],
    },
    Node {
        kind: PatternMatchAs,
        range: 55..62,
        source: `pattern`,
    }: {
        "leading": [],
        "dangling": [],
        "trailing": [
            SourceComment {
                text: "# 2",
                position: EndOfLine,
                formatted: true,
            },
            SourceComment {
                text: "# 3",
                position: OwnLine,
                formatted: true,
            },
            SourceComment {
                text: "# 4",
                position: EndOfLine,
                formatted: true,
            },
            SourceComment {
                text: "# 5",
                position: OwnLine,
                formatted: true,
            },
        ],
    },
}
```

Reply:
We'll need a custom handler in placement.rs for `MatchAs` that assigns all comments that come after the start of the name to the `MatchAs` clause.

Can you add a test case with a comment between the `as` and the name

```python
match pattern_comments:
    case (

        pattern 
        # 1
		as # 2
        # 3
		name #4
        # 5
    ):
        pass